modules/data.remote/R/phenology_MODIS_extract.R

```diff
@@ -0,0 +1,114 @@
+##' Get MODIS phenology data by date and location  
+##' 
+##' @export
+##' @param settings PEcAn settings
+##' @param run_parallel optional method to download data parallely. Only works if more than 1 
+##' site is needed and there are >1 CPUs available.
+##' @param ncores number of cpus to use if run_parallel is set to TRUE. If you do not know the 
+##' number of CPU's available, enter NULL. 
+##' 
+##' The output file will be saved to the folder defined in settings.
+##' 
+##' @examples
+##' \dontrun{
+##' settings <-  PEcAn.settings::read.settings("pecan.xml")
+##' test_modis_phenology <- PEcAn.data.remote::phenology_MODIS_extract(
+##'   settings = settings,
+##'   run_parallel = TRUE,
+##'   ncores = NULL)
+##' }
+##' @return a dataframe containing phenology dates and corresponding QA. 
+##' Output column names are "year", "site_id", "lat", "lon", "leafonday","leafoffday","leafon_qa","leafoff_qa"
+##' @author Qianyu Li
+##'
+
+phenology_MODIS_extract<- function(settings,run_parallel = TRUE,ncores = NULL){ 
```

Comment:
1. since this is an input processing function I'd recommend tweaking the function name and arguments to be more inline with existing input processing functions (e.g. starting with extract or download)
2. function doesn't seem to be called anywhere from within the existing workflow. I suggest adding to `do.conversions`
3. in keeping with the current philosophy of making PEcAn modules more independent, I'd recommend shifting the site_info bit into do.conversions itself and making the arguments to this function be (outdir, site_info, start_date, end_date, run_parallel, ncores). In other words, don't pass in the overall `settings` object, just pass the info you need.
4. Since this functionality is generally useful, not just in SDA, I'd recommend that the code you move to do.conversion grab start_date and end_date from the overall settings, not settings$state.data.assimilation